lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1225,19 +1224,17 @@ matchWitness(TypeChecker &tc,
 
     // Open up the witness type.
     witnessType = witness->getInterfaceType();
-    // FIXME: witness as a base locator?
```

Comment:
Previously consranits were added with nullptr locator, I've changed it to witness locator, I thought that's what FIXME was about, am I wrong here?
```cpp
locator = cs->getConstraintLocator(nullptr);
...
cs->addConstraint(ConstraintKind::Equal, reqType, witnessType, locator);
```